jms/runtime/src/main/java/io/quarkus/artemis/jms/runtime/camel/CamelContextEnhancer.java

```diff
@@ -0,0 +1,23 @@
+package io.quarkus.artemis.jms.runtime.camel;
+
+import java.util.Collections;
+import java.util.Map;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Observes;
+import javax.jms.ConnectionFactory;
+
+import org.apache.camel.impl.event.CamelContextInitializingEvent;
+
+import io.quarkus.artemis.core.runtime.ArtemisUtil;
+
+@ApplicationScoped
+public class CamelContextEnhancer {
+    void startUp(@Observes CamelContextInitializingEvent event) {
+        Map<String, ConnectionFactory> connectionFactoryNamesFromArc = ArtemisUtil
+                .extractIdentifiers(ConnectionFactory.class, Collections.emptySet());
+        for (var entry : connectionFactoryNamesFromArc.entrySet()) {
+            event.getContext().getRegistry().bind(entry.getKey(), entry.getValue());
```

Comment:
`camel-quarkus` should register these class automatically.

Reply:
You mean the beans? They are generated with `@Identifier(...)`, thus they are not registered in the context. I tried adding the `@Named(...)`-annotation, but this did not help either.

Reply:
@zhfeng Please see [this comment of mine](https://github.com/quarkiverse/quarkus-artemis/pull/122#discussion_r1045858434).

Reply:
@zhfeng can this discussion be resolved?

Reply:
I tried some additional things:
- adding `unremoveable()` to the synthetic `ConnectionFactory`-beans has no effect on the current behaviour
- adding dependency `camel-quarkus-bean` has no effect on the current behaviour